vendor/bat-native-ads/src/bat/ads/internal/notifications.cc

```diff
@@ -49,76 +49,91 @@ void Notifications::Initialize(InitializeCallback callback) {
 bool Notifications::Get(const std::string& id, NotificationInfo* info) const {
   DCHECK(is_initialized_);
 
-  auto notification = notifications_.find(id);
-  if (notification == notifications_.end()) {
+  auto iter = std::find_if(notifications_.begin(), notifications_.end(),
+      [&id](const auto& notification){return notification.id == id;});
+  if (iter == notifications_.end()) {
     return false;
   }
 
-  *info = notification->second;
+  *info = *iter;
 
   return true;
 }
 
-void Notifications::Add(const NotificationInfo& info) {
+void Notifications::PushBack(const NotificationInfo& info) {
   DCHECK(is_initialized_);
 
-  notifications_.insert({info.id, info});
+  notifications_.push_back(info);
+  ads_client_->ShowNotification(std::make_unique<NotificationInfo>(info));
 
   SaveState();
 }
 
-bool Notifications::Remove(const std::string& id) {
+void Notifications::PopFront(bool should_dismiss) {
+  if (!notifications_.empty()) {
+    if (should_dismiss){
+      ads_client_->CloseNotification(notifications_.front().id);
+    }
+    notifications_.pop_front();
+    SaveState();
+  }
+}
+
+bool Notifications::Remove(const std::string& id, bool should_dismiss) {
   DCHECK(is_initialized_);
 
-  if (!Exists(id)) {
+  auto iter = std::find_if(notifications_.begin(), notifications_.end(),
+      [&id](const auto& notification){return notification.id == id;});
+  if (iter == notifications_.end()) {
     return false;
   }
 
-  ads_client_->CloseNotification(id);
-  notifications_.erase(id);
+  if (should_dismiss){
+    ads_client_->CloseNotification(id);
+  }
+  notifications_.erase(iter);
 
   SaveState();
 
   return true;
 }
 
-void Notifications::RemoveAll() {
+void Notifications::RemoveAll(bool should_dismiss) {
```

Comment:
`should_dismiss` should be `const`

Reply:
I don't think so. I din't find a note about to pass values as constants in Google C++ code style .
Is there an example  in Chromium code?

Reply:
https://google.github.io/styleguide/cppguide.html#Use_of_const

Reply:
I wasn't sure, but the use of `const bool` does appear to be common in the chromium code and follows the style guide

Reply:
`Avoid const pass-by-value function parameters in function declarations`
"C++ Coding Standards" by Sutter, Alexandrescu.
15. Use const proactively.

Reply:
NONCRITICAL
Covered by: brave/brave-browser#6235 (11)

Reply:
@gdregalo We follow Chromium and if not specified fallback to Google style guides of `brave-browser` and `brave-core`